ctx.go

```diff
@@ -919,43 +913,55 @@ func (ctx *Ctx) SendFile(file string, compress ...bool) error {
 
 // SendStatus sets the HTTP status code and if the response body is empty,
 // it sets the correct status message in the body.
-func (ctx *Ctx) SendStatus(status int) {
-	ctx.Status(status)
+func (c *Ctx) SendStatus(status int) error {
+	c.Status(status)
+
 	// Only set status body when there is no response body
-	if len(ctx.Fasthttp.Response.Body()) == 0 {
-		ctx.SendString(utils.StatusMessage(status))
+	if len(c.fasthttp.Response.Body()) == 0 {
+		return c.SendString(utils.StatusMessage(status))
 	}
+
+	return nil
 }
 
 // SendString sets the HTTP response body for string types.
 // This means no type assertion, recommended for faster performance
-func (ctx *Ctx) SendString(body string) {
-	ctx.Fasthttp.Response.SetBodyString(body)
+func (c *Ctx) SendString(body string) error {
+	c.fasthttp.Response.SetBodyString(body)
+
+	return nil
 }
 
 // SendStream sets response body stream and optional body size.
-func (ctx *Ctx) SendStream(stream io.Reader, size ...int) {
+func (c *Ctx) SendStream(stream io.Reader, size ...int) error {
 	if len(size) > 0 && size[0] >= 0 {
-		ctx.Fasthttp.Response.SetBodyStream(stream, size[0])
+		c.fasthttp.Response.SetBodyStream(stream, size[0])
 	} else {
-		ctx.Fasthttp.Response.SetBodyStream(stream, -1)
-		ctx.Set(HeaderContentLength, strconv.Itoa(len(ctx.Fasthttp.Response.Body())))
+		c.fasthttp.Response.SetBodyStream(stream, -1)
+		c.setCanonical(HeaderContentLength, strconv.Itoa(len(c.fasthttp.Response.Body())))
 	}
+
+	return nil
 }
 
 // Set sets the response's HTTP header field to the specified key, value.
-func (ctx *Ctx) Set(key string, val string) {
-	ctx.Fasthttp.Response.Header.Set(key, val)
+func (c *Ctx) Set(key string, val string) {
+	c.fasthttp.Response.Header.Set(key, removeNewLines(val))
+	//c.fasthttp.Response.Header.Set(key, val)
```

Comment:
I'm going to stop leaving review comments on commented code as there appears to be quite a bit of it, and I don't wish to spam the PR with review comments.

Reply:
Oh please keep going, these are valuable reviews! I addressed all of them <3